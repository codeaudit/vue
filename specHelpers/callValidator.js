import ReactPropTypesArgs from './reactPropTypeArgs';

const callValidator = (validator, element, propName = '', componentName = '') => validator(element.props, propName, componentName, ...ReactPropTypesArgs);

export const validatorPasses = (validator, element, propName) => callValidator(validator, element, propName) === null;
export const validatorFails = (validator, element, propName) => callValidator(validator, element, propName) instanceof Error;
