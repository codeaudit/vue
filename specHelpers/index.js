import React from 'react';
import ThemeProvider, {getTheme} from './../src/Theme';

export const applyTheme = (ComponentUnderTest, theme = getTheme()) => <ThemeProvider theme={theme}>{ComponentUnderTest}</ThemeProvider>;
