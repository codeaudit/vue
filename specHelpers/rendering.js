import toJson from 'enzyme-to-json';
import {mount, shallow} from 'enzyme';
import React from 'react';
import theme from './TestTheme';

export const getRender = (Component) => (props = {}, context = {theme}) => shallow(<Component {...props} />, {context});
export const getMount = (Component) => (props = {}, context = {theme}) => mount(<Component {...props} />, {context});
export const snapshot = (renderedComponent) => toJson(renderedComponent);
