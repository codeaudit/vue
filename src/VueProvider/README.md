Enable application of a common theme across all Vue components and provide V1 functionality to all Vue components. This is essentially a combination of using [ThemeProvider](#/component/ThemeProvider) and [V1Provider](#/component/V1Provider).
