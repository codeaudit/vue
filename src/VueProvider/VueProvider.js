import React, {PropTypes} from 'react';
import {Provider} from 'react-redux';
import ThemeProvider from './../ThemeProvider';
import V1Provider from './../V1Provider';

const VueProvider = (props, context) => (
    <V1Provider runQuery={props.runQuery}>
        <ThemeProvider theme={props.theme}>
            <Provider store={context.store ? context.store : props.store}>
                {props.children}
            </Provider>
        </ThemeProvider>
    </V1Provider>
);
// VueProvider.displayName = 'VueProvider';
VueProvider.propTypes = {
    /**
     * Instance of the VersionOne JavaScript SDK
     */
    runQuery: PropTypes.func.isRequired,
    /**
     * A redux store with `ui` state segment property.
     */
    store: PropTypes.shape({
        dispatch: PropTypes.func.isRequired,
        getState: PropTypes.func.isRequired,
    }),
    /**
     * Theme to be used with Vue
     */
    theme: PropTypes.shape(ThemeProvider.themeDefinition).isRequired,
};
export default VueProvider;
