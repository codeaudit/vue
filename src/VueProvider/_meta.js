import examples from './examples';

export default {
    name: 'Vue Provider',
    keywords: ['theme', 'custom', 'css', 'style', 'stylesheet', 'visual', 'less', 'scss', 'sass', 'root', 'v1', 'meta', 'filter', 'token'],
    'see also': [],
    readme: require('./README.md'),
    status: 'stable',
    componentsSources: [
        {
            name: 'VueProvider',
            code: require('!raw!./VueProvider')
        }
    ],
    examples
};
