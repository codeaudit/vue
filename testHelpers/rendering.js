import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {mount, render, shallow} from 'enzyme';
import theme from './../packages/themes/src/V1Theme';

const ctx = {
    theme,
};

export const getShallow = (Component) => (props = {}, context = {}) => shallow(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getRender = (Component) => (props = {}, context = {}) => render(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getMount = (Component) => (props = {}, options = {}, context = {}) => mount(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
    ...options,
});

const replaceUIKey = (componentName) => {
    let index = 0;
    return (json) => {
        if (!json.props || !json.props.uiKey) {
            return json;
        }
        const uiKey = `${componentName}-${index}`;
        index += 1;
        return Object.assign({}, json, {
            uiKey,
            uiPath: [uiKey],
        });
    };
};
const replaceUiKeyRecursively = (json) => {
    if (!json.type) {
        return json;
    }
    const replaceUIKeyInParent = replaceUIKey(json.type);
    const output = replaceUIKeyInParent(json);
    if (!output.children) {
        return json;
    }
    output.children = json.children.map((child, index) => replaceUiKeyRecursively(child));
    return output;
};

export const snapshot = (renderedComponent) => {
    return toJson(renderedComponent);
};

export const reset = (component) => () => {
    if (component) {
        component.unmount();
    }
    document.body.innerHTML = '';
};
