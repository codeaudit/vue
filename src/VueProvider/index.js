import VueProvider from './VueProvider';

export default VueProvider;
