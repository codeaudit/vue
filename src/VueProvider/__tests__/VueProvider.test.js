import React from 'react';
import VueProvider from './../VueProvider';
import {getMount, snapshot} from './../../../specHelpers/rendering';

const shallowRender = getMount(VueProvider);

test('VueProvider provides the theme prop and V1 SDK prop to children via React\'s context', () => {
    const theme = getTheme();
    const v1 = getV1();
    const children = <span>Hello World</span>;

    const component = shallowRender({
        children,
        theme,
        v1,
    });
    expect(snapshot(component)).toMatchSnapshot();
});

// ---

function getTheme() {
    return {
        values: {
            _name: 'Test Theme',
            themeProperty: 'theme property value',
        },
    };
}

function getV1() {
    return {
        query: jest.fn(),
    };
}
