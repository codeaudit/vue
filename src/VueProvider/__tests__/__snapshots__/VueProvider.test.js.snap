exports[`test VueProvider provides the theme prop and V1 SDK prop to children via React's context 1`] = `
<VueProvider
  runQuery={[Function]}
  theme={
    Object {
      "values": Object {
        "_name": "Test Theme",
        "themeProperty": "theme property value",
      },
    }
  }>
  <V1Provider
    runQuery={[Function]}>
    <ThemeProvider
      theme={
        Object {
          "values": Object {
            "_name": "Test Theme",
            "themeProperty": "theme property value",
          },
        }
      }>
      <Provider>
        <span>
          Hello World
        </span>
      </Provider>
    </ThemeProvider>
  </V1Provider>
</VueProvider>
`;
